### PR TITLE
feat: switching to ESM-by-default & upgrading actions

### PR DESCRIPTION
**Pull request by dlockhart**
This accomplishes two things that both needed to happen together:

### 1) New Visual Diff dependencies

The main motivation for this change. By switching to the new `main` branch of the actions repo, you pick up the latest visual-diff, mocha@9 and puppeteer@12 -- this will likely case minor visual differences.

### 2) Make this project use ES Modules by default

Mocha 9 (required for the new visual-diff action) requires it, and web projects shouldn't be CommonJS anyway. I've converted the things using CommonJS things (like `require()` and `__dirname`) to their ESM equivalents.

### Other minor updates

While I was in here, I updated a few things like the latest dev-server and test-runner and linting configs.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,9 +21,8 @@ jobs:
         with:
           node-version: '14'
       - name: Semantic Release
-        uses: BrightspaceUI/actions/semantic-release@master
+        uses: BrightspaceUI/actions/semantic-release@main
         with:
-          DEFAULT_BRANCH: main
```
Comment (dlockhart) [inline .github/workflows/release.yml:26]: `main` is now the default.

File: demo/index.html [hunk 2]
```diff
@@ -2,7 +2,6 @@
 <html lang="en">
 	<head>
 		<link rel="stylesheet" href="/node_modules/@brightspace-ui/core/components/demo/styles.css" type="text/css">
-		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
```
Comment (dlockhart) [inline demo/index.html:5]: Don't need the polyfill anymore. 🎉

**Comment by svanherk** [comment]
Weird that only the one golden changed - and it does seem oddly aligned compared to the other goldens now, no? The unchanged goldens seem to have the text more middle-aligned

**Comment by dlockhart** [comment]
I'm not too sure why only that one changed, but inspecting the one with and without a label in my browser and then putting them side-by-side, the text alignment is identical:

Screen Shot 2021-12-06 at 5 06 20 PM

**Comment by svanherk** [comment]
Yeah, I can't reproduce the offset in the visual-diff page locally 😕

**Review by svanherk — APPROVED** [review]

**Review by jorge-ramirez-arredondo — APPROVED** [review]

**Pull request merged**